### PR TITLE
Addition of sidebar search filter for SideBarLink items

### PR DESCRIPTION
**Pull request by jazibobs**
I have updated `Sidebar.js` to include a search box which can quickly filter the `SideBarLink` items. I choose to make this addition directly in the Sidebar component as it is very closely tied with the `children` prop of `Sidebar.js`.

Additionally, I made a small change to the Examples page such that all direct children of the `Sidebar` component are `SidebarLink` components to ensure full compatibility with the new search feature.

Please test the search and let me know what you think, I hope you find it useful!

Issue: https://github.com/sButtons/sbuttons/issues/1446

### CODE REVIEW

File: website/src/components/Sidebar.js [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
 
 function Sidebar ({children = []}) {
+    const [filter, setFilter] = useState("");
+
     return (
         <div className="md:relative fixed animate__animated -left-full md:left-0 dark:bg-gray-700 bg-white shadow-md h-full md:block md:pt-0 md:w-4/12 top-0 w-9/12 z-20" id="sidebar">
             <div className="md:fixed h-full md:h-4/6 md:pt-0 md:w-4/12 overflow-y-auto pb-3 pl-5 pt-5">
-                {children}
+                <div className="flex pb-5">
+                    <label for="search" className="">Search:</label>
```
Comment (shahednasser) [inline website/src/components/Sidebar.js:10]: Remove the label and add "Search..." as a placeholder in the input

File: website/src/components/Sidebar.js [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
+                    <input 
+                        name="search" type="text"
+                        value={filter}
+                        onChange={(e) => setFilter(e.target.value)}
+                        className="ml-2 border-2 border-gray-400 focus:border-red-600"></input>
```
Comment (shahednasser) [inline website/src/components/Sidebar.js:15]: Change classes to `border focus:border-red-600 focus:outline-none pl-3`.

**Review by shahednasser — CHANGES_REQUESTED** [review]
Thank you for your work @jazibobs . The functionality is all good, I just have a few changes to the look and feel. Please find my comments below.

File: website/src/components/Sidebar.js (hunk 1 above)
Comment (jazibobs) [reply website/src/components/Sidebar.js]: This has been updated as requested.

File: website/src/components/Sidebar.js (hunk 2 above)
Comment (jazibobs) [reply website/src/components/Sidebar.js]: This has been updated as requested.

**Comment by shahednasser** [comment]
Thank you for your work!

@all-contributors please add @jazibobs for code

**Pull request merged**